consumer.go

```diff
@@ -231,6 +283,111 @@ func (c *Consumer) ScanShard(ctx context.Context, shardID string, fn ScanFunc) e
 	}
 }
 
+// ScanShardWithIntervalBatching waits for a specific second time interval to occur and then performs a batch fetch with specific shard, calls the callback func
+// with an array of records and checkpoints the scans
+
+func (c *Consumer) ScanShardWithIntervalBatching(ctx context.Context, shardID string, fn ScanFuncBatch) error {
+	// get last seq number from checkpoint
+
+	lastSeqNum, err := c.group.GetCheckpoint(c.streamName, shardID)
+	if err != nil {
+		return fmt.Errorf("get checkpoint error: %v", err)
+	}
+
+	// get shard iterator
+	shardIterator, err := c.getShardIterator(ctx, c.streamName, shardID, lastSeqNum)
+	if err != nil {
+		return fmt.Errorf("get shard iterator error: %v", err)
+	}
+
+	c.logger.Log("[CONSUMER] start scan:", shardID, lastSeqNum)
+	defer func() {
+		c.logger.Log("[CONSUMER] stop scan:", shardID)
+	}()
+	scanTicker := time.NewTicker(c.scanInterval)
+	defer scanTicker.Stop()
+	time_start := time.Now()
```

Comment:
if you run a linter on this code you'll see a bunch of warnings around CamelCase vs snake_case